src/Npgsql/ConnectorPool.Multiplexing.cs

```diff
@@ -68,8 +68,13 @@ async Task MultiplexingWriteLoop()
             Debug.Assert(_multiplexCommandReader != null);
 
             var timeout = _writeCoalescingDelayTicks / 2;
-            var timeoutTokenSource = new CancellationTokenSource();
-            var timeoutToken = timeout == 0 ? CancellationToken.None : timeoutTokenSource.Token;
+            var timeoutToken = CancellationToken.None;
+            TimeoutCancellationTokenSourceWrapper? timeoutTokenSource = null;
+            if (timeout != 0)
+            {
+                timeoutTokenSource = new TimeoutCancellationTokenSourceWrapper(TimeSpan.FromTicks(timeout));
```

Comment:
Note that this method is active for the lifetime of the program, so there's no need to avoid allocating TimeoutCancellationTokenSourceWrapper if no timeout is defined (this is why the code was as it was).